src/components/ListTemplate/listTemplate.js

```diff
@@ -0,0 +1,79 @@
+import React from 'react';
+
+import { IntlConsumer } from '@folio/stripes/core';
+
+import {
+  CheckboxColumn,
+  DefaultColumn,
+  MatchColumn,
+  TagsColumn,
+  DateColumn,
+} from './ColumnTemplates';
+import { formatUserName } from '../../utils';
+
+/**
+ * Retrieves and returns list of Column Templates renderProps
+ *
+ * @param {string} entityKey
+ * @param {string} searchTerm
+ * @param {object} [selectRecord]
+ * @param {array} [selectedRecords]
+ * @return {object} templates list
+ */
+export const listTemplate = (entityKey, searchTerm, selectRecord, selectedRecords) => ({
+  selected: record => (
+    <CheckboxColumn
+      value={record.id}
+      checked={selectedRecords.has(record.id)}
+      onChange={selectRecord}
+    />
+  ),
+  name: record => (
+    <DefaultColumn
+      iconKey={entityKey}
+      value={record.name}
+      searchTerm={searchTerm}
+    />
+  ),
+  match: record => (
+    <MatchColumn
+      record={record}
+      searchTerm={searchTerm}
+    />
+  ),
+  dataTypes: record => {
+    const { dataTypes } = record;
+
+    return (
+      <DefaultColumn value={Array.isArray(dataTypes) && dataTypes.length ? dataTypes.join(', ') : ''} />
```

Comment:
There is no need to check `dataTypes.length` as joining a zero-length array returns an empty string. And let's have it in one line with return.
```suggestion
      return <DefaultColumn value={Array.isArray(dataTypes) ? dataTypes.join(', ') : ''} />;
```